app/models/firmware_registry.rb

```diff
@@ -0,0 +1,37 @@
+class FirmwareRegistry < ApplicationRecord
+  include NewWithTypeStiMixin
+
+  has_many :firmware_binaries, :dependent => :destroy
+  has_one :endpoint, :as => :resource, :dependent => :destroy, :inverse_of => :resource
+  has_one :authentication, :as => :resource, :dependent => :destroy, :inverse_of => :resource
+
+  validates :name, :presence => true, :uniqueness => true
+
+  def sync_fw_binaries_queue
+    MiqQueue.put_unless_exists(
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => 'sync_fw_binaries'
```

Comment:
Any possibility this can take longer than 10 minutes to complete? The default queue timeout is 10 minutes.

Reply:
Thanks for pointing out, but we're just doing a single GET request to fetch a list of firmware metadata and store it in the vmdb, nothing more. I think we're safe timeout-wise.